lib/api/core/models/repositories/README.md

```diff
@@ -0,0 +1,225 @@
+# Kuzzle repositories
+
+The Repository classes allow to fetch and store some Kuzlle internal business object from/to the persistent layer.
+
+An abstract Repository class is available to be overriden if needed depending on the business needs.
+The class handles the persitence both to the CacheEngine (redis by default) and to the database (elastic search by default).
+
+The **Repository** class exposes the following methods:
+
+## Constructor
+
+The constructor expects to be passed the kuzzle instance and an options array.
+
+### options
+
+* *string* collection (mandatory)
+
+The underlying collection to store the Business objects to.
+
+* *integer | false* ttl (optional)
+
+The ttl to use for the CacheEngine. If set to false, the objects get an unlimited lifetime.
+
+* *function* ObjectConstructor (mandatory)
+
+The constructor function for the business objects. The load* methods will return some new instances of this function.
+
+* *function* readEngine (optional)
+
+The read engine to use to retrieve the business objects from the database. Defaults to Kuzzle's readEngine (elasticsearch).
+
+* *function* writeEngine (optional)
+
+The write engine to use to persist the business objects to the database. Defaults to Kuzzle's writeEngine (elasticsearch).
+
+* *function* cacheEngine (optional)
+
+The cache engine to use to store and retrieve the business objects to/from the cache. Defaults to Kuzzle's cacheEngine (redis).
+
+## loadOneFromDatabase
+
+```javascript
+Repository.prototype.loadOneFromDatabase = function (id) {...}
+```
+
+This method loads one object from the database given its id.
+
+### parameters
+
+* *string* id
+
+The id of the document to retrieve.
+
+### returns
+
+Returns a promise that resolves to the *ObjectConstructor* instance if found or null if the id does not exist in the database.
+
+## loadMultiFromDatabase
+
+```javascript
+Repository.prototype.loadMultiFromDatabase = function (ids) {...}
+```
+
+This method tries to load the business objects form the database matching the given ids.
+
+### returns
+
+Returns a promise that resolves to an array containing the *ObjectConstructor* instances that could be retrieved from the database.
+
+If no matching document could be found, an emtpy array is returned.
+
+## loadFromCache
+
+```javascript
+Repository.prototype.loadFromCache = function (id, opts) {...}
+```
+
+This method loads one object from the cache given its id.
+
+### parameters
+
+* *string* id
+
+The object id to load from the cache.
+
+* *Object* opts
+
+An optional options object.
+
+Currently, the only optional parameter that can be pass to the method id *key*. If no key is given to the method, the key used by the method defaults to *collection* + '/' + id.
```

Comment:
typo : "the only optional parameter that can be pass to the method IS *key*" (instead of "id key") ??